script.js

```diff
@@ -0,0 +1,116 @@
+const form = document.querySelector("#survey-form");
+const resultsDiv = document.querySelector("#results");
+
+//Question and answer object array
+const questions = [
+    {
+        question: "What is your favorite color?",
+        answers: [
+            { text: "Red", value: "red" },
+            { text: "Green", value: "green" },
+            { text: "Blue", value: "blue" }
+        ]
+    },
+    {
+        question: "What is your favorite food?",
+        answers: [
+            { text: "Pizza", value: "pizza" },
+            { text: "Sushi", value: "sushi" },
+            { text: "Steak", value: "steak" }
+        ]
+    },
+    {
+        question: "What is your favorite pet?",
+        answers: [
+            { text: "Dog", value: "dog" },
+            { text: "Cat", value: "cat" },
+            { text: "Bird", value: "bird" }
+        ]
+    }
+];
+let answersArray = [];
+let responseFrequency = [];
+
+const renderForm = () =>
+{
+    form.innerHTML += `<h2 class="is-size-3 my-5">Survey Questions:</h2>`;
+    questions.forEach((question) =>
+    {
+        form.innerHTML += `<h3 class="is-size-5 my-4">${question.question}</h3>`;
+        question.answers.forEach((answer) =>
+        {
+            form.innerHTML += `
+          <input class="radio" type="radio" id="${answer.value}" name="${question.question}" value="${answer.value}">
+          <label class="radio" for="${answer.value}">${answer.text}</label><br>
+        `;
+        });
+    });
+    form.innerHTML += `<button class="button is-medium my-6" type="submit">Submit</button>`;
+};
+
+renderForm(); //End of form render
+
+form.addEventListener("submit", (event) =>
+{
+    event.preventDefault();
+    const name = form.elements.name.value;
+    let answers = [];
+    let responses = [];
+
+    questions.forEach((question) =>
+    {
+        const selectedOption = form.elements[question.question].value;
+        answers.push({ [question.question]: selectedOption });
+        responses.push(selectedOption);
+    });
+
+    responseFrequency.push(responses);
+    // console.log(responseFrequency);
+
+    answersArray.push({ name, answers });
+
+    let result = "";
+    result += `<p class="is-size-3 my-4">All Responses and report:</p>`;
+    result += `<ul>`;
+    answersArray.forEach((person) =>
+    {
+        result += `<li class="is-size-6 mb-2 mt-5">${person.name} responses were:</li>`;
+        person.answers.forEach((answer) =>
+        {
+            for (const question in answer) {
+                result += `<ul class="is-size-6 ml-6">
+                <li>For the question: ${question} the response was: ${answer[question]}</li>
+                </ul>`;
+            }
+        });
+    });
+
+    result += `</ul>`;
+
+    resultsDiv.innerHTML = result;
+
+    //Finding the frequency of responses
```

Comment:
you need to find the report related things only after all the survey submission. otherwise each time it will call the all methods and ineffect reduce the efficiency. refactor it as generating only after survey submissions and clicking any button for generate reports